RecoLocalTracker/SiPixelRecHits/src/PixelCPEClusterRepair.cc

```diff
@@ -0,0 +1,628 @@
+      if ( theClusterParam.with_track_angle )
+      {
+         theClusterParam.templXrec_ = theDetParam.theTopol->localX( theClusterParam.theCluster->x(), theClusterParam.loc_trk_pred ) - lorentz_drift * micronsToCm; // rough Lorentz drift correction
+         theClusterParam.templYrec_ = theDetParam.theTopol->localY( theClusterParam.theCluster->y(), theClusterParam.loc_trk_pred );
+      }
+      else
+      {
+         edm::LogError("PixelCPEClusterRepair")
+         << "@SUB = PixelCPEClusterRepair::localPosition"
+         << "Should never be here. PixelCPEClusterRepair should always be called with track angles. This is a bad error !!! ";
+         
+         theClusterParam.templXrec_ = theDetParam.theTopol->localX( theClusterParam.theCluster->x() ) - lorentz_drift * micronsToCm; // rough Lorentz drift correction
+         theClusterParam.templYrec_ = theDetParam.theTopol->localY( theClusterParam.theCluster->y() );
+      }
+   }   
+   else 
+   {
+      //--- Template Reco succeeded.
+      theClusterParam.hasFilledProb_ = true;
+
+      //--- Go from microns to centimeters
+      theClusterParam.templXrec_ *= micronsToCm;
+      theClusterParam.templYrec_ *= micronsToCm;
+      
+      //--- Go back to the module coordinate system
+      theClusterParam.templXrec_ += lp.x();
+      theClusterParam.templYrec_ += lp.y();
+   }
+   return;
+}
+
+
+
+
+//------------------------------------------------------------------
+//  localError() relies on localPosition() being called FIRST!!!
+//------------------------------------------------------------------
+LocalError
+PixelCPEClusterRepair::localError(DetParam const & theDetParam,  ClusterParam & theClusterParamBase) const
+{
+   
+   ClusterParamTemplate & theClusterParam = static_cast<ClusterParamTemplate &>(theClusterParamBase);
+   
+   //cout << endl;
+   //cout << "Set PixelCPETemplate errors .............................................." << endl;
+   
+   //cout << "CPETemplate : " << endl;
+   
+   //--- Default is the maximum error used for edge clusters.
+   //--- (never used, in fact: let comment it out, shut up the complains of the static analyzer, and save a few CPU cycles)
+   //   const float sig12 = 1./sqrt(12.0);
+   //   float xerr = theDetParam.thePitchX *sig12;
+   //   float yerr = theDetParam.thePitchY *sig12;
+   float xerr = 0.0f, yerr = 0.0f;
+   
+   // Check if the errors were already set at the clusters splitting level
+   if ( theClusterParam.theCluster->getSplitClusterErrorX() > 0.0f && theClusterParam.theCluster->getSplitClusterErrorX() < 7777.7f &&
+       theClusterParam.theCluster->getSplitClusterErrorY() > 0.0f && theClusterParam.theCluster->getSplitClusterErrorY() < 7777.7f )
+   {
+      xerr = theClusterParam.theCluster->getSplitClusterErrorX() * micronsToCm;
+      yerr = theClusterParam.theCluster->getSplitClusterErrorY() * micronsToCm;
+      
+      //cout << "Errors set at cluster splitting level : " << endl;
```

Comment:
Please remove the commented out lines, or replace with some LogDebug or similar

Reply:
OK, if you require this change now.